src/Data/Time/PreciseDuration.purs

```diff
@@ -52,46 +55,60 @@ unPreciseDuration = case _ of
   Weeks d -> d
 
 -- Each duration in nanoseconds
```

Comment:
Perhaps this stuff should be hidden in a new module `Data.Time.PreciseDuration.Internal`?